### PR TITLE
Fixes #3836. SetupFakeDriver sometimes causes failure in the unit test.

### LINKED ISSUE #3836: SetupFakeDriver sometimes causes failure in the unit test.
Yes, because sometimes it causes failure in the `With_Subview_Using_PosFunc` unit test. See https://github.com/gui-cs/Terminal.Gui/actions/runs/11903293490/job/33169963032. The reason is a racing condition where the `SetupFakeDriver` was resizing the terminal to (5,5) and when the above unit test is running the terminal size isn't what it's expected and corrupt the expected value.

```
  Failed Terminal.Gui.LayoutTests.DimAutoTests.With_Subview_Using_PosFunc [3 ms]
  Error Message:
   Assert.Equal() Failure: Values differ
Expected: 25
Actual:   20
  Stack Trace:
     at Terminal.Gui.LayoutTests.DimAutoTests.With_Subview_Using_PosFunc() in /home/runner/work/Terminal.Gui/Terminal.Gui/UnitTests/View/Layout/Dim.AutoTests.PosTypes.cs:line 647
   at System.RuntimeMethodHandle.InvokeMethod(Object target, Void** arguments, Signature sig, Boolean isConstructor)
   at System.Reflection.MethodBaseInvoker.InvokeWithNoArgs(Object obj, BindingFlags invokeAttr)
```

_Originally posted by @BDisp in https://github.com/gui-cs/Terminal.Gui/pull/3828#discussion_r1848887686_

### PR DESCRIPTION
**Pull request by BDisp**
## Fixes

- Fixes #3836

## Proposed Changes/Todos

- [x] `Application.Screen` was causing this issue because `_screen` wasn't being set to null after running a test using the `SetupFakeDriver`.

## Pull Request checklist:

- [x] I've named my PR in the form of "Fixes #issue. Terse description."
- [x] My code follows the [style guidelines of Terminal.Gui](https://github.com/gui-cs/Terminal.Gui/blob/develop/.editorconfig) - if you use Visual Studio, hit `CTRL-K-D` to automatically reformat your files before committing.
- [x] My code follows the [Terminal.Gui library design guidelines](https://github.com/gui-cs/Terminal.Gui/blob/develop/CONTRIBUTING.md)
- [x] I ran `dotnet test` before commit
- [ ] I have made corresponding changes to the API documentation (using `///` style comments)
- [x] My changes generate no new warnings
- [x] I have checked my code and corrected any poor grammar or misspellings
- [x] I conducted basic QA to assure all features are working

### CODE REVIEW

**Comment by BDisp** [comment]
This error doesn't happens all the time. It's a racing condition when a unit test that use the `SetupFakeDriver` after it's finished and another unit test that doesn't use it start and will call the `superView.SetRelativeLayout (new (10, 10));` when the `Application.Screen` was already be set to that size before in the first unit test.

File: Terminal.Gui/Application/Application.Initialization.cs [hunk 1]
```diff
@@ -163,10 +163,7 @@ internal static void InternalInit (
                                                 );
         }
 
-        Driver.SizeChanged += Driver_SizeChanged;
-        Driver.KeyDown += Driver_KeyDown;
-        Driver.KeyUp += Driver_KeyUp;
-        Driver.MouseEvent += Driver_MouseEvent;
+        InitState ();
```
Comment (tig) [inline Terminal.Gui/Application/Application.Initialization.cs:166]: 1) Please rename this to `SubscribeDriverEvents`
2) Move the call up to line 153 (just after the call to `Driver!.init()`)
3) Add another `internal` method `UnsubscribeDriverEvents` that does the `-=` and replace the code in `Application.cs` line 180 with a call to this new method.

**Review by tig — CHANGES_REQUESTED** [review]

File: Terminal.Gui/Application/Application.Initialization.cs [hunk 2]
```diff
@@ -176,6 +172,24 @@ internal static void InternalInit (
...
+
+        Driver.SizeChanged += Driver_SizeChanged;
+        Driver.KeyDown += Driver_KeyDown;
+        Driver.KeyUp += Driver_KeyUp;
+        Driver.MouseEvent += Driver_MouseEvent;
+    }
+
+    internal static void UnsubscribeDriverEvents ()
+    {
+        Driver.SizeChanged -= Driver_SizeChanged;
```
Comment (tig) [inline Terminal.Gui/Application/Application.Initialization.cs:187]: Add `ArgumentNullException.ThrowIfNull (Driver);` here

**Review by tig — CHANGES_REQUESTED** [review]

File: Terminal.Gui/Application/Application.Initialization.cs (hunk 2 above)
Comment (BDisp) [reply Terminal.Gui/Application/Application.Initialization.cs:189]: Good catch. Done.

**Review by tig — APPROVED** [review]
Well done. Thanks.

**Pull request merged**